File: src/main/java/com/mtisma/ppp/model/Product.java
package com.mtisma.ppp.model;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    private Long id;

    @ManyToOne
    private Category category;

    @NotNull
    private String name;

    private BigDecimal currentPrice;

    @Lob
    private String description;

    @OneToMany(mappedBy = "product", cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    private List<Specification> specifications;

    @OneToMany(mappedBy = "product")
    private List<Image> images;
}

File: frontend/src/types/product.ts

export interface Category {
  id: number,
  name: string
}

export interface Product {
  id: number,
  category: Category,
  name: string,
  images: Image[],
  currentPrice: number,
  description: string,
  specifications: Specification[]
}

export interface Specification {
  name: string,
  value: string
}

export interface PriceHistory {
  amount: number,
  createdAt: Date
}

export interface Image {
  id: number
}